### PR TITLE
feat: improve local development experience

### PR DESCRIPTION
**Pull request by oliverbaehler**
Fixes some stuff I have experienced while developing locally:

- Correct GinkGO Path
- Exclude capsule bin with gitignore
- Split E2E Testing Recipes into three stages. e2e/* now always deletes the kind cluster. e2e-exec runs jsut ginkgo without rebuilding the cluster
- Correct Admission Webhooks Patches (Mutating path was incorrect, Validating had wrong order)
- Add documentation to fetch tags

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -134,19 +134,19 @@ dev-setup:
...
-			{'op': 'replace', 'path': '/webhooks/6/clientConfig', 'value':{'url':\"$${WEBHOOK_URL}/services\",'caBundle':\"$${CA_BUNDLE}\"}},\
-			{'op': 'replace', 'path': '/webhooks/7/clientConfig', 'value':{'url':\"$${WEBHOOK_URL}/tenants\",'caBundle':\"$${CA_BUNDLE}\"}},\
-			{'op': 'replace', 'path': '/webhooks/8/clientConfig', 'value':{'url':\"$${WEBHOOK_URL}/nodes\",'caBundle':\"$${CA_BUNDLE}\"}}\
+			{'op': 'replace', 'path': '/webhooks/4/clientConfig', 'value':{'url':\"$${WEBHOOK_URL}/nodes\",'caBundle':\"$${CA_BUNDLE}\"}},\
+			{'op': 'replace', 'path': '/webhooks/5/clientConfig', 'value':{'url':\"$${WEBHOOK_URL}/pods\",'caBundle':\"$${CA_BUNDLE}\"}},\
+			{'op': 'replace', 'path': '/webhooks/6/clientConfig', 'value':{'url':\"$${WEBHOOK_URL}/persistentvolumeclaims\",'caBundle':\"$${CA_BUNDLE}\"}},\
+			{'op': 'replace', 'path': '/webhooks/7/clientConfig', 'value':{'url':\"$${WEBHOOK_URL}/services\",'caBundle':\"$${CA_BUNDLE}\"}},\
+			{'op': 'replace', 'path': '/webhooks/8/clientConfig', 'value':{'url':\"$${WEBHOOK_URL}/tenants\",'caBundle':\"$${CA_BUNDLE}\"}}\
```
Comment (prometherion) [inline Makefile:149]: The webhooks order seems wrong to me, are you sure it is correct?

File: Makefile [hunk 2]
```diff
@@ -236,5 +239,9 @@ e2e/%: ginkgo
 		--set 'manager.readinessProbe.failureThreshold=10' \
 		capsule \
 		./charts/capsule
+
+e2e-exec:
 	$(GINKGO) -v -tags e2e ./e2e
-	kind delete cluster --name capsule
+
+e2e-destroy:
+	kind delete cluster --name capsule
```
Comment (prometherion) [inline Makefile:247]: Missing new line at the end of the file.

File: docs/content/contributing/development.md [hunk 3]
```diff
@@ -115,6 +115,12 @@ upstream	https://github.com/clastix/capsule.git (fetch)
 upstream	https://github.com/clastix/capsule.git (push)
 ```
 
+Pull all tags
+
+```
+$ git pull upstream
```
Comment (prometherion) [inline docs/content/contributing/development.md:121]: Shouldn't be it with `git fetch --all && git pull upstream`?

File: .gitignore [hunk 4]
```diff
@@ -30,3 +30,4 @@ bin
 .DS_Store
 *.tgz
 
+capsule
```
Comment (prometherion) [inline .gitignore:33]: Missing new line at the end of the file.

**Review by prometherion — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (oliverbaehler) [reply Makefile:149]: Yes, that's the correct order at least for everywhere I tested:

```
kubectl get validatingwebhookconfigurations.admissionregistration.k8s.io capsule-validating-webhook-configuration -o=jsonpath="{range .webhooks[*]}{.name}{'\n'}{end}"

cordoning.tenant.capsule.clastix.io
ingress.capsule.clastix.io
namespaces.capsule.clastix.io
networkpolicies.capsule.clastix.io
nodes.capsule.clastix.io
pods.capsule.clastix.io
pvc.capsule.clastix.io
services.capsule.clastix.io
tenants.capsule.clastix.io
```

**Review by prometherion — APPROVED** [review]

**Pull request merged**